### DISCUSSION TITLE
Estraxt all pages as images

### DISCUSSION THREAD

**Discussion by denisgottardello** [qpdf/qpdf#934] [General]
Hi, is there a way to extract all pages as images?
Can anyone share the code?
Best regards.

**Comment by jberkenbilt** [comment]
It would be more converting pages to images than extracting as PDF pages are not generally stored as images. qpdf can't do this, but ghostscript can, as can ImageMagick and GraphicsMagick, which both run ghostscript under the covers.

Example: to render pages as 200 DPI png files with ghostscript:

```
gs -q -r200 -dNOPAUSE -sDEVICE=png16m -dBATCH -sOutputFile=out-%d.png in.pdf
```
You can write other file formats. Tiff files can contain multiple pages but are not commonly used by people who are not doing image work, though they are very efficient for pure black and white with g4 compression. You could use jpeg, etc. I'll refer you to ghostscript documentation for more details.

**Comment by jberkenbilt** [comment]
You can also extract images from PDF pages, but that's not what you asked. Tools pdfimages from poppler can do that.

**Comment by denisgottardello** [comment]
Hi and many thanks for your replies. I need the c++ code (and not a command line) to have a image from a pdf page using qpdf library.
Have you got anything to share?

Best regards.
